### ISSUE TITLE
About the attribute ”training“ of the model

### ISSUE THREAD

**Issue by Kylinlhy** [roytseng-tw/Detectron.pytorch#104]
I found that there is a override function "train(self, mode=True)"in lib/modeling/Resnet.py. I want to know the reason why you override the function.
Besides, I found a problem when inference. When inference, after setting model.eval(), the attribute "training" should be False, but when I print the value of the attribute, I found sometimes the value is True, which will cause a problem after adding the Cross-GPU BatchNorm module.

**Comment by wwtalwtaw** [comment]
That override function is used to set the batch norm of freezing layers under evaluation mode.

But the problem is that the author didn't set batch norm layers under evaluation mode explicitly. The override function results in the fact that these batch norm are always in training mode.

In my opinion, the override function should be like the following:
```python
    def train(self, mode=True):
        # Override
        self.training = mode

        for i in range(1, cfg.RESNETS.FREEZE_AT + 1):
            getattr(self, 'res%d' % i).eval()

        for i in range(cfg.RESNETS.FREEZE_AT + 1, self.convX + 1):
            getattr(self, 'res%d' % i).train(mode)
```
